bpf/bpf_lxc.c

```diff
@@ -588,11 +624,11 @@ static inline int __inline__ ipv6_policy(struct __sk_buff *skb, int ifindex, __u
 	tuple.nexthdr = ip6->nexthdr;
 	ipv6_addr_copy(&tuple.addr, (union v6addr *) &ip6->saddr);
 	l4_off = ETH_HLEN + ipv6_hdrlen(skb, ETH_HLEN, &tuple.nexthdr);
+	csum_l4_offset_and_flags(tuple.nexthdr, &off);
 
 	/* derive reverse NAT index and zero it. */
 	rev_nat_index = ip6->daddr.s6_addr32[3] & 0xFFFF;
-	if (rev_nat_index) {
-		int csum_off, csum_flags = 0;
+	if ((skb->cb[CB_FLAGS] & CB_F_CLUSTER_DST) && rev_nat_index) {
```

Comment:
I will drop this patch as it was only needed to test LB to endpoints in containers with an address outside of our cluster.